packages/edit-post/src/editor.js

```diff
@@ -127,6 +126,9 @@ function Editor( {
 			// This is marked as experimental to give time for the quick inserter to mature.
 			__experimentalSetIsInserterOpened: setIsInserterOpened,
 			keepCaretInsideBlock,
+			// Keep a reference of the `allowedBlockTypes` from the server to handle use cases
+			// where we need to differentiate if a block is disabled by the user or some plugin.
+			defaultAllowedBlockTypes: settings.allowedBlockTypes,
```

Comment:
This is not needed right? Or am I missing something?

Reply:
This seems to be needed because below we override the `allowedBlockTypes` in the editor settings taking into account the `hiddenBlockTypes`.

Reply:
Was this a bug in trunk, I fail to see how this related to the removal of the context provider.

Reply:
It's not a bug, but when `BlockManager` was introduced the `allowedBlockTypes` was used in combination with the new `hiddenBlockTypes` setting. This decision still makes sense to me, but the alternative would be to leave `allowedBlockTypes` intact, and take into account `hiddenBlockTypes` in every place we would need - mainly in `Inserter` logic.

Makes sense?

Reply:
yes, sorry I got confused :)